src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -632,6 +632,12 @@ private bool AcquireServerCredentials(ref byte[] thumbPrint, byte[] clientHello)
             X509Certificate localCertificate = null;
             bool cachedCred = false;
 
+            // There are three options for selecting the server certificate. When 
+            // selecting which to use, we prioritize the new ServerCertSelectionDelegate
+            // API. If the new API isn't used we call LocalCertSelectionCallback, and if 
+            // neither is set we fall back to using ServerCertificate. When choosing
+            // from multiple certificates ServerCertSelectionDelegate is the preferred
+            // API. LocalCertSelectionCallback is maintained only for compat reasons.
```

Comment:
If ServerCertSelectionDelegate returns null, should we use ServerCertificate if it was provided? (I'm actually asking; I'm not sure if there's a use case for that or not.)

Reply:
Is "ServerCertificate" the parameter passed to the public API [AuthenticateAsServer](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasserver?view=netcore-2.0) method?  If so, then yes, we should use that.

Reply:
Look at the two options, either you throw the NotSupportedException like the framework does and may gain simpler support for LCSCB OR you default to the cert from the constructor and regain some of the functionality you lost by removing  localCertificates from the arguments of the callback (falling back to initial cert, localCertificates[0] == ServerCertificate). But it's not enough, you still don't have the initial certificate available in the callback (a way to solve this would be a public property on SslStream that returns this initial certificate). 
So no, it's probably better to go with framework behavior here.

Reply:
We should go primarily after .NET Framework compat. Let's throw the same exception and make sure we have a test covering the scenario.

Reply:
And the test exists: SslStream_ServerCallbackReturnsNull_Throws